src/DynamicRupture/Output/Base.cpp

```diff
@@ -214,77 +216,329 @@ void Base::updateElementwiseOutput() {
+
+      auto& slipVectors = std::get<VariableID::Slip>(outputData.vars);
+      if (slipVectors.isActive) {
+        VrtxCoords crossProduct = {0.0, 0.0, 0.0};
+        MeshTools::cross(strike, tangent1, crossProduct);
+
+        double cos1 = MeshTools::dot(strike, tangent1);
+        double scalarProd = MeshTools::dot(crossProduct, normal);
+
+        // Note: cos1**2 can be greater than 1.0 because of rounding errors -> min
+        double sin1 = std::sqrt(1.0 - std::min(1.0, cos1 * cos1));
+        sin1 = (scalarProd > 0) ? sin1 : -sin1;
+
+        auto* slip1 = local.layer->var(drDescr->slip1);
+        auto* slip2 = local.layer->var(drDescr->slip2);
+
+        slipVectors(DirectionID::Strike, level, i) =
+            cos1 * slip1[local.ltsId][local.nearestGpIndex] -
+            sin1 * slip2[local.ltsId][local.nearestGpIndex];
+
+        slipVectors(DirectionID::Dip, level, i) = sin1 * slip1[local.ltsId][local.nearestGpIndex] +
+                                                  cos1 * slip2[local.ltsId][local.nearestGpIndex];
+      }
+      this->outputSpecifics(outputData, level, i);
     }
 
-    auto& peakSlipsRate = std::get<VariableID::PeakSlipRate>(outputData.vars);
-    if (peakSlipsRate.isActive) {
-      peakSlipsRate(level, i) = peakSR[ltsId][nearestGaussPoint];
+    if (type == OutputType::AtPickpoint) {
+      outputData.cachedTime[outputData.currentCacheLevel] = time;
+      outputData.currentCacheLevel += 1;
     }
+  }
+}
+
+void Base::computeLocalStresses() {
+  real pFactorPlus = local.waveSpeedsPlus->pWaveVelocity * local.waveSpeedsPlus->density;
+  real pFactorMinus = local.waveSpeedsMinus->pWaveVelocity * local.waveSpeedsMinus->density;
+
+  real sFactorPlus = local.waveSpeedsPlus->sWaveVelocity * local.waveSpeedsPlus->density;
+  real sFactorMinus = local.waveSpeedsMinus->sWaveVelocity * local.waveSpeedsMinus->density;
+
+  real norDivisor = 1.0 / (pFactorMinus + pFactorPlus);
+  real shearDivisor = 1.0 / (sFactorMinus + sFactorPlus);
+  real uVelDivisor = 1.0 / (pFactorPlus);
+
+  auto diff = [this](int i) {
+    return this->local.faceAlignedValuesMinus[i] - this->local.faceAlignedValuesPlus[i];
+  };
+
+  local.xyStress = local.faceAlignedValuesPlus[3] +
+                   ((diff(3) + sFactorMinus * diff(7)) * sFactorPlus) * shearDivisor;
+
+  local.xzStress = local.faceAlignedValuesPlus[5] +
+                   ((diff(5) + sFactorMinus * diff(8)) * sFactorPlus) * shearDivisor;
+
+  local.p = local.faceAlignedValuesPlus[0] +
+            ((diff(0) + pFactorMinus * diff(6)) * pFactorPlus) * norDivisor;
+
+  local.u =
+      local.faceAlignedValuesPlus[6] + (local.p - local.faceAlignedValuesPlus[0]) * uVelDivisor;
+
+  real missingSigmaValues = (local.p - local.faceAlignedValuesPlus[0]);
+  missingSigmaValues *= (1.0 - 2.0 * std::pow(local.waveSpeedsPlus->sWaveVelocity /
+                                                  local.waveSpeedsPlus->pWaveVelocity,
+                                              2));
```

Comment:
same: Where is this formula is the current master branch? I would like to double-check it.